File: PepCodes/Recursion_05Oct/recursionArray.cpp
#include <iostream>
#include <vector>
using namespace std;

void displayArray(vector<int> &arr, int vidx) {

    if(vidx+1 == arr.size()) return ;

    cout<<arr[vidx] << " ";
    displayArray(arr, vidx+1);

}

bool find(vector<int> &arr, int vidx, int data)  {
    
    if(arr[vidx] == data) {
        return true;
    }
    if (vidx==arr.size())
    {
     return false;
    }
    
    return find(arr, vidx+1, data);

}

//int maximum(vector<int> &arr, int vidx) { }
//int minimum(vector<int> &arr, int vidx) {}
//int lastIndex(vector<int> &arr, int vidx, int data) {}

//vector<int> allIndex(vector<int> &arr, int vidx, int data, ) {} //do in java





int main(int args, char ** argv) {

    vector<int> arr = {10, 6, 8, 10, 4, 5, 5, 6, 8, -3, 2, 12, 8, 3};
    // displayArray(arr,0);

    cout<<find(arr,0, -31)<<endl;

    return 0;
}

File: PepCodes/Recursion_05Oct/recursionArray.java
public class recursionArray {

    public static void main(String[] args) {
        System.out.println(stepCases(5));
    }

//    public static int[] allIndex(int[] arr, int vidx,int data, boolean flag) {
//
//        if(flag) {
//           for(int n: arr) {
//                if(data == n) { 
//                    size++; 
//                }
//            }
//            return new int[size];
//        }
//    }

  //  public static int[] arrLength(int[] arr, data) {
  //      int size =0;
  //      for(int n : arr) {
  //          if(n == data) size++;
  //       }
  //      return new int[size];
  //  }


  public static int stepCases(int n) {
      if(n ==1 || n==0) {
          return 1;
      } 
      if(n == 2) return 2;
      if(n == 3) return 3;

      return stepCases(n-1) + stepCases(n-2) +stepCases(n-3);
  }
} 